tests/test_io.py

```diff
@@ -1899,3 +1899,142 @@ def test_get_variables_data_values4(self):
         for index, test_data in test_data_sets.items():
             np.testing.assert_array_almost_equal(test_data[0], reference_data['time'][index])
             np.testing.assert_array_almost_equal(test_data[1], reference_data['@Diagnostics.nbr_steps'][index])
+
+if assimulo_installed:
+    class TestFileSizeLimit:
+
+        def _setup(self, result_type, result_file_name="", fmi_type="me"):
+            if fmi_type == "me":
+                model = Dummy_FMUModelME2([], os.path.join(file_path, "files", "FMUs", "XML", "ME2.0", "CoupledClutches.fmu"), _connect_dll=False)
+            else:
+                model = Dummy_FMUModelCS2([], os.path.join(file_path, "files", "FMUs", "XML", "CS2.0", "CoupledClutches.fmu"), _connect_dll=False)
+                
+            opts = model.simulate_options()
+            opts["result_handling"]  = result_type
+            opts["result_file_name"] = result_file_name
+
+            return model, opts
+
+        def _test_result(self, result_type, result_file_name="", max_size=1e6):
+            model, opts = self._setup(result_type, result_file_name)
+
+            opts["result_max_size"] = max_size
+
+            #No exception should be raised.
+            res = model.simulate(options=opts)
+
+        def _test_result_exception(self, result_type, result_file_name="", fmi_type="me"):
+            model, opts = self._setup(result_type, result_file_name, fmi_type)
+
+            opts["result_max_size"] = 10
+
+            with nose.tools.assert_raises(ResultSizeError):
+                res = model.simulate(options=opts)
+
+        def _test_result_size_verification(self, result_type, result_file_name=""):
+            model, opts = self._setup(result_type, result_file_name)
+
+            max_size = 1e6
+            opts["result_max_size"] = max_size
+            opts["ncp"] = 10000
+
+            with nose.tools.assert_raises(ResultSizeError):
+                res = model.simulate(options=opts)
+
+            result_file = model.get_last_result_file()
+            if result_file:
+                file_size = os.path.getsize(result_file)
+
+                assert file_size > max_size*0.9 and file_size < max_size*1.1, \
+                        "The file size is not within 10% of the given max size"
+        """
```

Comment:
```suggestion
        # TODO: Pytest parametrization
        """
```
Pytest parametrizations will reduce this to 2 tests + parametrization decorators, so a lot more compact.